wazimap_ng/profile/serializers/indicator_data_serializer.py

```diff
@@ -104,49 +111,20 @@ def rearrange_group(group_dict):
                     }
         return group_dict
 
-    def sort_group_subindicators(row, group_dict):
-        new_dict = {}
-        for group, group_subindicators_dict in group_dict.items():
-            key = (group, row["dataset"])
-            if key in groups_lookup:
-                key_func = lambda x: x[0]
-                subindicator_order = groups_lookup[key]
-                sorted_group_subindicators_list = sort_list_using_order(group_subindicators_dict.items(), subindicator_order, key_func=key_func)
-                sorted_group_subindicators_dict = OrderedDict(sorted_group_subindicators_list)
-            else:
-                logger.warning(f"Key: {key} not in groups lookup")
-                sorted_group_subindicators_dict = group_subindicators_dict
-
-            new_dict[group] = sorted_group_subindicators_dict
-
-        return new_dict
-
-    def sort_indicator_subindicators(row, group_dict):
-        key = (row["indicator_group"][0], row["dataset"])
-        key_func = lambda x: x[0]
-
-        new_group_dict = {}
-        for group, group_subindicators_dict in group_dict.items():
-            new_group_subindicators_dict = {}
-            for group_subindicator, indicator_subindicators_dict in group_subindicators_dict.items():
-                if key in groups_lookup:
-                    subindicator_order = groups_lookup[key]
-                    items = indicator_subindicators_dict.items()
-                    sorted_tuples = sort_list_using_order(items, subindicator_order, key_func=key_func)
-                    sorted_indicator_subindicators_dict = OrderedDict(sorted_tuples)
-                else:
-                    sorted_indicator_subindicators_dict = indicator_subindicators_dict
-                new_group_subindicators_dict[group_subindicator] = sorted_indicator_subindicators_dict
-            new_group_dict[group] = new_group_subindicators_dict
-
-        return new_group_dict
-
     def prepare_json(row):
-        json_data = rearrange_group(row["jsdata"]["groups"])
-        json_data = sort_group_subindicators(row, json_data)
-        json_data = sort_indicator_subindicators(row, json_data)
+        dataset = row["dataset"]
+        groups = row["indicator_group"]
+        primary_group = groups[0]
+        sorter = sorters[dataset]
+
+        group_data = row["jsdata"]["groups"]
+        group_data = rearrange_group(group_data)
+        subindicators = row["jsdata"]["subindicators"]
+
+        group_data = sorter.sort_groups(group_data, primary_group)
+        row["jsdata"]["subindicators"] = sorter.sort_subindicators(primary_group, subindicators)
```

Comment:
is that an expected side-effect that you are manipulating the input with explicitly returning it? 

since you're returning data and use that returned data it feels strange that you can manipulate the input.

Reply:
Yes - that was pre-existing. I don't like it but I didn't want to fiddle too much in this PR. The next bit of work will address that.